R/model.R

```diff
@@ -270,22 +274,43 @@ CmdStanModel <- R6::R6Class(
       }
       if (!is.null(stan_file) && compile) {
         self$compile(...)
-      }
-      if (length(self$exe_file()) > 0 && file.exists(self$exe_file())) {
-        cpp_options <- model_compile_info(self$exe_file())
-        for (cpp_option_name in names(cpp_options)) {
-          if (cpp_option_name != "stan_version" &&
-              (!is.logical(cpp_options[[cpp_option_name]]) || isTRUE(cpp_options[[cpp_option_name]]))) {
-            private$cpp_options_[[cpp_option_name]] <- cpp_options[[cpp_option_name]]
+      } else {
+        # set exe path, same logic as in compile
+        if(!is.null(private$dir_)){
+          dir <- repair_path(absolute_path(private$dir_))
+          assert_dir_exists(dir, access = "rw")
+          if (length(self$exe_file()) != 0) {
+            self$exe_file(file.path(dir, basename(self$exe_file())))
+          }
+        }
+        if (length(self$exe_file()) == 0) {
+          if (is.null(private$dir_)) {
+            exe_base <- self$stan_file()
+          } else {
+            exe_base <- file.path(private$dir_, basename(self$stan_file()))
+          }
+          self$exe_file(cmdstan_ext(strip_ext(exe_base)))
+          if (dir.exists(self$exe_file())) {
+            stop("There is a subfolder matching the model name in the same folder as the model! Please remove or rename the subfolder and try again.", call. = FALSE)
```

Comment:
This code block is hit when either no stan file is provided or compile is set to false.

If there's a folder that has the name that the executable is suppose do to have...indeed we have no executable and also we can't compile to create one because the folder is in the way.

If no stan file is provided...basically that means the user just gave us an exe_file path, and that path is a path to a directory, not an executable, so there's no way we can recover from that. We don't have a binary or a stan file.

If compile is set to false but a stan file is provided...this is more like a warning that both operations that call the binary will fail because there's no binary there. Also, compiling will fail because we can't put the binary where it belongs.


I believe with the status quo, this issue is ignored when the object is instantiated, and we just with for the user to try compiling or sampling or what have you and then fail at that point.

Reply:
Thanks, that sounds reasonable. 

> I believe with the status quo, this issue is ignored when the object is instantiated, and we just with for the user to try compiling or sampling or what have you and then fail at that point.

I think that's probably right

Reply:
If you agree with my reasoning, I think no changes are needed to my PR. If you disagree, please un-resolve this comment.